test/strings/types.jl

```diff
@@ -163,27 +163,29 @@ rs = RevString("foobar")
 @test rsplit(RevString("ailuj"),'l') == ["ju","ia"]
 @test parse(Float64,RevString("64")) === 46.0
 
-# reverseind
-for T in (String, UTF16String, UTF32String)
-    for prefix in ("", "abcd", "\U0001d6a4\U0001d4c1", "\U0001d6a4\U0001d4c1c", " \U0001d6a4\U0001d4c1")
-        for suffix in ("", "abcde", "\U0001d4c1β\U0001d6a4", "\U0001d4c1β\U0001d6a4c", " \U0001d4c1β\U0001d6a4")
-            for c in ('X', 'δ', '\U0001d6a5')
-                s = convert(T, string(prefix, c, suffix))
-                ri = search(reverse(s), c)
-                @test reverse(s) == RevString(s)
-                @test c == s[reverseind(s, ri)] == reverse(s)[ri]
-                s = RevString(s)
-                ri = search(reverse(s), c)
-                @test c == s[reverseind(s, ri)] == reverse(s)[ri]
-                s = convert(T, string(prefix, prefix, c, suffix, suffix))
-                pre = convert(T, prefix)
-                sb = SubString(s, nextind(pre, endof(pre)), endof(convert(T, string(prefix, prefix, c, suffix))))
-                ri = search(reverse(sb), c)
-                @test c == sb[reverseind(sb, ri)] == reverse(sb)[ri]
-            end
-        end
-    end
-end
+# # reverseind
+# for T in (String, GenericString)
```

Comment:
> -            out += 4

Could we replace "4"  with a macro that says what it is? That way we
know which "4"'s  we must replace when on the day a string uses more
digits. It will happen, it happened in C. (Used to be 1, then 2,  then
4)

Because eventually there will be 8 or 16,  for some reason we do not
realize yet.  The same is true for numbers which mean something,  and
will need to be replaced. Better it is replaced in a macro once,
rather than searching through the entire library code. Let us not have
a "2000" bug waiting in the code. It is also more readable.


On Thu, Jun 30, 2016 at 12:36 PM, Steven G. Johnson <
notifications@github.com> wrote:

> In test/strings/types.jl
> <https://github.com/JuliaLang/julia/pull/16590#discussion_r69166495>:
>
> > -                @test reverse(s) == RevString(s)
> > -                @test c == s[reverseind(s, ri)] == reverse(s)[ri]
> > -                s = RevString(s)
> > -                ri = search(reverse(s), c)
> > -                @test c == s[reverseind(s, ri)] == reverse(s)[ri]
> > -                s = convert(T, string(prefix, prefix, c, suffix, suffix))
> > -                pre = convert(T, prefix)
> > -                sb = SubString(s, nextind(pre, endof(pre)), endof(convert(T, string(prefix, prefix, c, suffix))))
> > -                ri = search(reverse(sb), c)
> > -                @test c == sb[reverseind(sb, ri)] == reverse(sb)[ri]
> > -            end
> > -        end
> > -    end
> > -end
> > +# # reverseind
> > +# for T in (String, GenericString)
>
> I don't see how reverseind breaks any abstraction. Given an index to a
> codepoint in reverse(s), it converts it to an index in s. This operation
> is well-defined regardless of the encoding, and regardless of what the
> indices look like.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaLang/julia/pull/16590/files/8a007860c0594ad60858b6127b2764b12c8b37aa#r69166495>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ATKyItXdzSuCkFvkrI7F1vLHWCsWTuZKks5qQ_CqgaJpZM4InB5w>
> .
>